autoPyTorch/pipeline/components/training/trainer/__init__.py

```diff
@@ -403,7 +415,8 @@ def early_stop_handler(self, X: Dict[str, Any]) -> bool:
         if self.checkpoint_dir is None:
             self.checkpoint_dir = tempfile.mkdtemp(dir=X['backend'].temporary_directory)
 
-        epochs_since_best = self.run_summary.get_last_epoch() - self.run_summary.get_best_epoch()
+        epochs_since_best = self.run_summary.get_last_epoch() -\
+            self.run_summary.get_best_epoch(dataset=self.early_stopping_dataset)
```

Comment:
Since this line calls two methods, I think it is better to separate the line so that we can easily track errors when we get an error.
```suggestion
        last_epoch = self.run_summary.get_last_epoch()
        best_epoch = self.run_summary.get_best_epoch(dataset=self.early_stopping_dataset)
        epochs_since_best =  last_epoch - best_epoch            
```